src/p2wsh.js

```diff
@@ -14,11 +16,140 @@
+ * scriptPubkey and the script pubkey itself
+ * @param {Number} [scriptPubkeySize = 34] size of script pubkey. 
+ * Defaults to 34 which is the size of a P2WSH script pubkey and the
+ * largest possible standard
+ * @returns {Number} size of tx output (default: 43)
+ */
+function txoutSize(scriptPubkeySize = 34) {
+  // per_output: value (8) + script length (1) + 
+  const VAL_BYTES = 8;
+  const scriptLengthBytes = encoding.sizeVarint(scriptPubkeySize);
+  // for P2WSH Locking script which is largest possible(34)
+  return VAL_BYTES + scriptLengthBytes + scriptPubkeySize;
+}
+
+/**
+ * @description calculates size of redeem script given n pubkeys.
+ * Calculation looks like: 
+ * OP_M (1 byte) + size of each pubkey in redeem script (OP_DATA= 1 byte * N) +
+ * pubkey size (33 bytes * N) + OP_N (1 byte) + OP_CHECKMULTISIG (1 byte)
+ *  => 1 + (1 * N) + (33 * N) + 1 + 1
+ * @param {Number} n - value of n in m-of-n for multisig script
+ * @returns {Number} 3 + 34 * N
+ */
+export function getRedeemScriptSize(n) {
+  const OP_M_BYTES = 1;
+  const OP_N_BYTES = 1;
+  const opDataBytes = n; // 1 byte per pubkey in redeem script
+  const pubkeyBytes = 33 * n;
+  const OP_CHECKMULTISIG_BYTES = 1;
+  return OP_M_BYTES + opDataBytes + pubkeyBytes + OP_N_BYTES + OP_CHECKMULTISIG_BYTES;
+}
+
+/**
+ * @description Calculates the value of a multisig witness given m-of-n values
+ * Calculation is of the following form:
+ * witness_items count (varint 1+) + null_data (1 byte) + size of each signature (1 byte * OP_M) + signatures (73 * M) +
+ * redeem script length (1 byte) + redeem script size (4 + 34 * N bytes)
+ * @param {Number} m - value of m in m-of-n for multisig script
+ * @param {Number} n - value of n in m-of-n for multisig script
+ * @returns {Number} 6 + (74 * M) + (34 * N)
+ */
+export function getWitnessSize(m, n) {
+  const OP_NULL_BYTES = 1; // needs to be added b/c of bug in multisig implementation
+  const opDataBytes = m;
+  // assumes largest possible signature size which could be 71, 72, or 73
+  const signaturesSize = 73 * m;
+  const REDEEM_SCRIPT_LENGTH = 1;
+  const redeemScriptSize = getRedeemScriptSize(n);
+  // total witness stack will be null bytes + each signature (m) + redeem script
+  const WITNESS_ITEMS_COUNT = encoding.sizeVarint(1 + m + 1);
+  
+  return WITNESS_ITEMS_COUNT + 
+    OP_NULL_BYTES + 
+    opDataBytes + 
+    signaturesSize + 
+    REDEEM_SCRIPT_LENGTH + 
+    redeemScriptSize;
+}
+
+/**
+ * @description Calculates the size of the fields in a transaction which DO NOT
+ * get counted towards witness discount.
+ * Calculated as: version bytes (4) + locktime bytes (4) + input_len (1+) + txins (41+) + output_len (1+) + outputs (9+)
+ * @param {*} inputsCount - number of inputs in the tx
+ * @param {*} outputsCount - number of outputs in the tx
+ * @returns {Number} number of bytes in the tx without witness fields
+ */
+export function calculateBase(inputsCount, outputsCount) {
```

Comment:
If used in other contexts (namely P2SH-P2WSH/P2WPKH) this would need to be paramtarized for larger txins, but other fields are the same.